lasagne/layers/normalization.py

```diff
@@ -101,3 +109,234 @@ def get_output_for(self, input, **kwargs):
+        The nonlinearity that is applied to the layer activations. If None
+        is provided, the layer will be linear (this is the default).
+    mode : {'low_mem', 'high_mem'}
+        Specify which batch normalization implementation to use: ``'low_mem'``
+        avoids storing intermediate representations and thus requires less
+        memory, while ``'high_mem'`` can reuse representations for the backward
+        pass and is thus 5-10% faster.
+    beta : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\beta`. Must match
+        the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    gamma : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\gamma`. Must
+        match the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    mean : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\mu`. Must match
+        the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    var : Theano shared variable, expression, numpy array, or callable
+        Initial value, expression or initializer for :math:`\\sigma^2`. Must
+        match the incoming shape, skipping all axes in `axes`.
+        See :func:`lasagne.utils.create_param` for more information.
+    **kwargs
+        Any additional keyword arguments are passed to the :class:`Layer`
+        superclass.
+
+    Notes
+    -----
+    This layer should be inserted between a linear transformation (such as a
+    :class:`DenseLayer`, or :class:`Conv2DLayer`) and its nonlinearity. The
+    convenience function :func:`batch_norm` modifies an existing layer to
+    insert batch normalization in front of its nonlinearity.
+
+    The behavior can be controlled by passing keyword arguments to
+    :func:`lasagne.layers.get_output()` when building the output expression
+    of any network containing this layer.
+
+    During training, [1]_ normalize each input mini-batch by its statistics
+    and update an exponential moving average of the statistics to be used for
+    validation. This can be achieved by passing ``deterministic=False``.
+    For validation, [1]_ normalize each input mini-batch by the stored
+    statistics. This can be achieved by passing ``deterministic=True``.
+
+    For more fine-grained control, ``batch_norm_update_averages`` can be passed
+    to update the exponential moving averages (``True``) or not (``False``),
+    and ``batch_norm_use_averages`` can be passed to use the exponential moving
+    averages for normalization (``True``) or normalize each mini-batch by its
+    own statistics (``False``). These settings override ``deterministic``.
+
```

Comment:
I am getting the following error while trying to use the code:

  File "/home/dedigadot/Envs/env3/local/lib/python2.7/site-packages/Lasagne-0.2.dev1-py2.7.egg/lasagne/layers/normalization.py", line 179, in __init__
    if axis not in self.shared_axes]

AttributeError: 'BatchNormLayer' object has no attribute 'shared_axes'


And also, in lines:
        beta = beta.dimshuffle(pattern)
        gamma = gamma.dimshuffle(pattern)

Souldn't they be:
        self.beta = beta.dimshuffle(pattern)
        self.gamma = gamma.dimshuffle(pattern)